experimental/sdata/sdata.go

```diff
@@ -0,0 +1,226 @@
+package sdata
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+// Less sure about this interface.
+// Also, probably would need to make a type with methods that wrap the interface so methods can be added
+// without breaking changes - if we even want an interface like this
+// But for now helps illustrate at least
+type TimeSeriesCollection interface {
+	AddMetric(metricName string, l data.Labels, t []time.Time, values interface{}) error
+	SetMetricMD(metricName string, l data.Labels, fc data.FieldConfig)
+	AppendMetricValue(metricName string, l data.Labels, t time.Time, value interface{}) error
+	InsertMetricValue(metricName string, l data.Labels, t time.Time, value interface{}) error
+	Validate() (isEmpty bool, errors []error)
+	AsWideFrameSeries() *WideFrameSeries
+	AsMultiFrameSeries() *MultiFrameSeries
+}
+
+func ValidValueFields() []data.FieldType {
+	return append(data.NumericFieldTypes(), []data.FieldType{data.FieldTypeBool, data.FieldTypeNullableBool}...)
+}
+
+// or perhaps a container struct with non-exported fields (for indicies and such) and the Frames exported.
+type MultiFrameSeries []*data.Frame
+
+// values must be a numeric slice such as []int64, []float64, []*float64, etc or []bool / []*bool.
+func (mfs *MultiFrameSeries) AddMetric(metricName string, l data.Labels, t []time.Time, values interface{}) error {
+	var err error
+
+	if !data.ValidFieldType(values) {
+		return fmt.Errorf("type %T is not a valid data frame field type", values)
+	}
+
+	valueField := data.NewField(metricName, l, values) // note
```

Comment:
^^ worth checking the details on this.  I *think* it encourages a pattern where we copy/allocate the same data 3x+

vs the flavors where you know the length upfront and set at each index.  In that case we only allocate once (+ whatever the initial data was)

Reply:
You mean the implementation of data.NewField? (Just checking that you didn't mean to comment on AppendMetricValue below)

Reply:
Hmm... look at the implementation, not sure why the vector can not be passed as a whole when the vector is created, there is `newInt8Vector`, not not one that takes the type an asserts it... I can perhaps look into that.

Reply:
exactly -- assuming we have a typed go struct (not interface{}) I think the type assertions are not necessary. And obviously avoiding a 2nd copy will be helpful :)